flexmeasures/api/common/utils/validators.py

```diff
@@ -186,16 +203,18 @@ def wrapper(fn):
         @wraps(fn)
         @as_json
         def decorated_service(*args, **kwargs):
-            form = get_form_from_request(request)
-            if form is None:
-                current_app.logger.warning(
-                    "Unsupported request method for unpacking 'duration' from request."
-                )
-                return invalid_method(request.method)
+            # TODO: marshmallow doesn't support timestamps in iso8601 str representation
```

Comment:
It seems we would have to write our own Marshmallow type, which is not too hard. It would be nice of course for having an API which gives you proper responses about input. But we'd need a proper way to validate a ISO6801 duration string, like a Regex. Do you know where that stuff lives?

Reply:
Nice work!